src/Symfony/Bundle/FrameworkBundle/Command/ContainerDebugCommand.php

```diff
@@ -176,19 +179,21 @@ protected function getContainerBuilder()
             return $this->containerBuilder;
         }
 
-        if (!$this->getApplication()->getKernel()->isDebug()) {
-            throw new \LogicException('Debug information about the container is only available in debug mode.');
-        }
-
-        if (!is_file($cachedFile = $this->getContainer()->getParameter('debug.container.dump'))) {
-            throw new \LogicException('Debug information about the container could not be found. Please clear the cache and try again.');
+        $kernel = $this->getApplication()->getKernel();
+        $cacheDir = $kernel->getCacheDir();
+        $containerClass = $kernel->getContainer()->getParameter('kernel.container_class');
+        $cache = new ConfigCache($cacheDir.'/'.$containerClass.'.xml', true);
+
+        if (!$cache->isFresh()) {
+            $buildContainer = \Closure::bind(function () { return $this->buildContainer(); }, $kernel, get_class($kernel));
+            $container = $buildContainer();
+            $container->getCompilerPassConfig()->setRemovingPasses(array());
+            $container->compile();
+            $cache->write((new XmlDumper($container))->dump(), array(new FileResource($cacheDir.'/'.$containerClass.'.php')));
```

Comment:
you should rather register the ContainerBuilder resources here rather than a FileResource for the compiled container